src/components/documentsList.js

```diff
@@ -1,115 +1,115 @@
-import React, { Component } from 'react';
-import {StyleSheet, Text, View, FlatList, Dimensions as ScreenDimensions, Image } from 'react-native';
-import boundMethod from 'autobind-decorator';
-import Dimensions from '../constants/dimensions';
-import Colors from '../constants/colors';
+import React, { useEffect } from "react";
+import { Text, View, FlatList, Image } from "react-native";
+import { Dimensions as ScreenDimensions, StyleSheet } from "react-native";
+import Colors from "../constants/colors";
+import Dimensions from "../constants/dimensions";
 
-let width = ScreenDimensions.get('window').width;
+import withFireBase from "./withFirebase";
 
-const rcaImageuri =  'https://gdb.rferl.org/DE8BE3A3-E1D5-4C0F-B9DC-AB4A92A23B06_cx0_cy9_cw0_w1023_r1_s.png';
-const headerImageUri =  'https://aa-boschbcs-by.resource.bosch.com/media/_tech/images/backgrounds/visual_workshopfinder.jpg';
+const headerImageUri =
+  "https://aa-boschbcs-by.resource.bosch.com/media/_tech/images/backgrounds/visual_workshopfinder.jpg";
 
-const data = [
-  { key: '1', uri: rcaImageuri, type: 'ITP', expire: '21-12-2012' },
-  { key: '2', uri: rcaImageuri, type:'rca', expire:' 21-12-2012'},
-  { key: '3', uri: rcaImageuri, type:'rca', expire:' 21-12-2012'},
-  { key: '4', uri: rcaImageuri , type:'taxa de drum', expire:' 21-12-2012'},
-  { key: '5', uri: rcaImageuri, type:'rca', expire:' 21-12-2012'},
-  { key: '6', uri: rcaImageuri , type:'rca', expire:' 21-12-2012'},
-  { key: '7', uri: rcaImageuri  , type:'rca', expire:' 21-12-2012'}
-];
-
-const formatData = (data, numColumns) => {
+const getDocuments = (data, numColumns) => {
   const numberOfFullRows = Math.floor(data.length / numColumns);
 
-  let numberOfElementsLastRow = data.length - (numberOfFullRows * numColumns);
-  while (numberOfElementsLastRow !== numColumns && numberOfElementsLastRow !== 0) {
-    data.push({ key: `blank-${numberOfElementsLastRow}`, empty: true });
-    numberOfElementsLastRow++;
+  // Add the mandatory key for each item
+  // displayed in the FlatList.
+  data.map((item, index) => (item.key = index));
+
+  let numberOfElementsLastRow = data.length - numberOfFullRows * numColumns;
+  if (numberOfElementsLastRow) {
+    while (numberOfElementsLastRow !== numColumns) {
+      data.push({ key: `blank-${numberOfElementsLastRow}`, empty: true });
+      ++numberOfElementsLastRow;
+    }
   }
   return data;
 };
 
-class DocumentsList extends Component {
-  renderItem = ({ item, index }) => {
+const DocumentsList = props => {
+  // Fetch data from Firebase.
+  useEffect(() => {
+    props.fetchData();
+  }, []);
+
+  // Render function used by FlatList.
+  const renderItem = item => {
```

Comment:
```suggestion
  const renderItem = ({item}) => {
```
And you could get rid off next line

Reply:
I actually can't .. because the item is inside item .. lol :)